components/hab/src/cli/hab/pkg.rs

```diff
@@ -437,3 +425,130 @@ pub enum Pkg {
+        }
+    }
+
+    pub fn args(&self) -> &[OsString] {
+        if self.format_and_args.is_empty() {
+            &[]
+        } else {
+            &self.format_and_args[1..]
+        }
+    }
+}
+
+/// This struct can only be constructed within this module because it has a private field. All
+/// `ExportFormat` variants contain an instance of this types. This effectively "seals" construction
+/// of `ExportFormat` to this module. Construction should only happen in `FromStr` which uses
+/// conditional compilation to only allow constructing platform supported `ExportFormat`s. By
+/// "sealing" construction of this enum we avoid the need to litter conditional compilation wherever
+/// `ExportFormat` is used.
+pub struct SealedExportFormat(());
+
+pub enum ExportFormat {
+    Cf(SealedExportFormat),
+    Container(SealedExportFormat),
+    Docker(SealedExportFormat),
+    Mesos(SealedExportFormat),
+    Tar(SealedExportFormat),
+}
+
+impl ExportFormat {
+    const ALL_FORMATS: &'static [&'static str] = &[Self::CF,
+                                                   Self::CONTAINER,
+                                                   Self::DOCKER,
+                                                   Self::MESOS,
+                                                   Self::TAR];
+    const CF: &'static str = "cf";
+    const CONTAINER: &'static str = "container";
+    const DOCKER: &'static str = "docker";
+    const MESOS: &'static str = "mesos";
+    const TAR: &'static str = "tar";
+
+    fn possible_values() -> String {
+        // Filter formats that are not supported by the current platform
+        let platform_supported_formats =
+            Self::ALL_FORMATS.iter()
+                             .filter(|format| ExportFormat::from_str(format).is_ok())
+                             .copied()
+                             .collect::<Vec<_>>();
+        platform_supported_formats.join(", ")
+    }
+}
+
+impl Display for ExportFormat {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Cf(_) => write!(f, "{}", Self::CF),
+            Self::Container(_) => write!(f, "{}", Self::CONTAINER),
+            Self::Docker(_) => write!(f, "{}", Self::DOCKER),
+            Self::Mesos(_) => write!(f, "{}", Self::MESOS),
+            Self::Tar(_) => write!(f, "{}", Self::TAR),
+        }
+    }
+}
+
+impl FromStr for ExportFormat {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            #[cfg(any(target_os = "linux", target_os = "windows"))]
+            Self::CF => Ok(Self::Cf(SealedExportFormat(()))),
+            #[cfg(any(target_os = "linux", target_os = "windows"))]
+            Self::CONTAINER => Ok(Self::Container(SealedExportFormat(()))),
+            #[cfg(any(target_os = "linux", target_os = "windows"))]
+            Self::DOCKER => Ok(Self::Docker(SealedExportFormat(()))),
```

Comment:
Since "docker" is an old deprecated alias of "container", could we just not have `Self::Docker` and map both strings to `Self::Container`?

Reply:
Great suggestion! Addressed in [d748fba](https://github.com/habitat-sh/habitat/pull/7857/commits/d748fbabef54092b166042e7f704d160e542bb82)